metadata-ingestion/docs/sources/snowflake/snowflake_pre.md

```diff
@@ -99,6 +99,33 @@ The steps slightly differ based on which you decide to use.
   including `client_id` and `client_secret`, plus your Okta user's `Username` and `Password`
   * Note: the `username` and `password` config options are not nested under `oauth_config`
 
+### Snowflake Shares
+If you are using [Snowflake Shares](https://docs.snowflake.com/en/user-guide/data-sharing-provider) to share data across different snowflake accounts, and you have set up DataHub recipes for ingesting metadata from all these accounts, you may end up having multiple similar dataset entities corresponding to virtual versions of same table in different snowflake accounts. DataHub Snowflake connector can automatically link such tables together through Siblings and Lineage relationship if user provides information necessary to establish the relationship using configurations `inbound_shares_map` and `outbound_shares_map` in recipe. 
+
+#### Example
+- Snowflake account `account1` (ingested as platform_instance `instance1`) owns a database `db1`. A share `X` is created in `account1` that includes database `db1` along with schemas and tables inside it. 
+- Now, `X` is shared with snowflake account `account2` (ingested as platform_instance `instance2`). A database `db1_from_X` is created from inbound share `X` in `account2`.
+- In this case, all tables and views included in share `X` will also be present in `instance2`.`db1_from_X`. You would need following configurations in snowflake recipe to setup Siblings and Lineage relationships correctly.
+- In snowflake recipe of `account1` :
+
+  ```yaml
+  account_id: account1
+  platform_instance: instance1
+  outbound_shares_map:
+      db1: 
+          - platform_instance: instance2 # this is a list, as db1 can be shared with multiple snowflake accounts using X
+            database_name: db1_from_X
+  ```
+- In snowflake recipe of `account2` :
```

Comment:
This has crossed my mind too and I'd really like "having a config that could be the same for every recipe". The only downside is we require addition information - e.g. share name "X" and it is possible that some of the shares config will not be relevant for some account recipes, making validation and probable errors hard to find. I feel, the ease of using same shares config outweighs the downsides so let me think more on this and update.

Reply:
I've updated the config to use similar structure as your example, except that using term `consumers` instead of `outbounds`. Outbound is relative and can be confusing term. Consumer has a peculiar meaning for snowflake shares and hence unambiguous.

Reply:
If we don't want to have them specify the share name, we can also do:
```
  shares:
    - platform_instance: instance1
      database_name: db1
      consumers:
        - platform_instance: instance2 # this is a list, as db1 can be shared with multiple snowflake accounts using X
          database_name: db1_from_X
    - platform_instance: ...
```
but maybe that's more confusing

Reply:
yeah, I like the one with share name better - more precise and readable. Also shares do have unique names across accounts. This change primarily makes shares configuration absolute and exhaustive for all accounts and configurations need not be thought of wrt the particular account/recipe.